File: leetcode/blind-150/test4.cs
// https://leetcode.com/problems/group-anagrams/description/

public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {

        var groups = new Dictionary<string, IList<string>>();
        foreach (string item in strs)
        {
            var sum = new char[26];
            foreach (char c in item)
            {
                sum[c - 'a']++;
            }

            var key = new string(sum);

            if (!groups.ContainsKey(key))
            {
                groups[key] = new List<string>();
            }
            groups[key].Add(item);
        }
        return groups.Values.ToList();
    }
}

File: leetcode/blind-150/test4.py
# https://leetcode.com/problems/group-anagrams/description/

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        groups = {}
        for item in strs:
            key = [0] * 26
            for c in item:
                key[ord(c) - ord('a')] += 1

            tkey = tuple(key)
            if tkey not in groups.keys():
                groups[tkey] = []

            groups[tkey].append(item)

        return groups.values()
